src/samples/audio/detect_test.c

```diff
@@ -136,12 +142,13 @@ static void notify_kpb(const struct comp_dev *dev)
 		       sizeof(cd->event_data));
 }
 
-static void detect_test_notify(const struct comp_dev *dev)
+void detect_test_notify(const struct comp_dev *dev)
 {
```

Comment:
This should be in a separate patch where you explain why this should be available to outside modules. In fact, I think all the changes from detect_test.c should be in separate patches with a proper explanation of they are needed.